polygeist/lib/Dialect/Polygeist/Transforms/BareMemRefToLLVM.cpp

```diff
@@ -21,6 +21,24 @@ using namespace mlir;
 using namespace mlir::polygeist;
 
 namespace {
+/// Returns the LLVM type of the global variable given the memref type `type`.
+///
+/// Copied from `mlir/lib/Conversion/MemRefToLLVM/MemRefToLLVM.cpp`
+static Type convertGlobalMemrefTypeToLLVM(MemRefType type,
+                                          LLVMTypeConverter &typeConverter) {
+  // LLVM type for a global memref will be a multi-dimension array. For
+  // declarations or uninitialized global memrefs, we can potentially flatten
+  // this to a 1D array. However, for memref.global's with an initial value,
+  // we do not intend to flatten the ElementsAttribute when going from std ->
+  // LLVM dialect, so the LLVM type needs to me a multi-dimension array.
+  Type elementType = typeConverter.convertType(type.getElementType());
+  Type arrayTy = elementType;
+  // Shape has the outermost dim at index 0, so need to walk it backwards
```

Comment:
The comment is a bit confusing, probably better to say row/column major (even it can be more than 2, that would make things clearer)

Reply:
I have the same opinion as @whitneywhtsang. Probably not worth going through the upstreaming process, so we can keep as is